src/Microsoft.FileFormats/PE/PEFile.cs

```diff
@@ -105,6 +106,55 @@ public bool IsValid()
         /// </summary>
         public ImageDataDirectory ComDataDirectory { get { return ImageDataDirectory[(int)ImageDirectoryEntry.ComDescriptor]; } }
 
+        /// <summary>
+        /// Returns the address of a module export symbol if found
+        /// </summary>
+        /// <param name="symbolName">symbol name (without the module name prepended)</param>
+        /// <param name="offset">symbol offset returned</param>
+        /// <returns>true if found</returns>
+        public bool TryGetExportSymbol(string symbolName, out ulong offset)
+        {
+            try
+            {
+                ImageExportDirectory exportDirectory = _exportDirectory.Value;
+                if (exportDirectory is not null)
+                {
+                    for (int nameIndex = 0; nameIndex < exportDirectory.NumberOfNames; nameIndex++)
+                    {
+                        uint namePointerRVA = RelativeVirtualAddressReader.Read<uint>((ulong)(exportDirectory.AddressOfNames + (sizeof(uint) * nameIndex)));
+                        if (namePointerRVA != 0)
+                        {
+                            string name = RelativeVirtualAddressReader.Read<string>(namePointerRVA);
+                            if (name == symbolName)
+                            {
+                                ushort ordinalForNamedExport = RelativeVirtualAddressReader.Read<ushort>((ulong)(exportDirectory.AddressOfNameOrdinals + (sizeof(ushort) * nameIndex)));
+                                offset = RelativeVirtualAddressReader.Read<uint>((ulong)(exportDirectory.AddressOfFunctions + (sizeof(uint) * ordinalForNamedExport)));
```

Comment:
This does not consider the case of a forward RVA export, thought that is an edge case that might not apply to our case. If `offset` is within the export section, then according to https://docs.microsoft.com/en-us/windows/win32/debug/pe-format#export-address-table it's a pointer to a string that forwards the export to another dll potentially.

Reply:
Can we check `ordinalForNamedExport` < `NumberOfFunctions`?

Reply:
The runtime PEDecoder doesn't check for either a forward RVA export or if `ordinalForNamedExport < NumberOfFunctions`.